packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -275,35 +296,28 @@ class Auth {
 
     /**
      * WARNING: This function is relevant to be used in Hybrid deployments only.
-     * Compares the refresh_token keys for guest('cc-nx-g') and registered('cc-nx') login from the cookie received from SFRA with the copy stored in localstorage on PWA Kit
-     * to determine if the login state of the shopper on SFRA site has changed. If the keys are different we return true considering the login state did change. If the keys are same,
-     * we compare the values of the refresh_token to cover an edge case where the login state might have changed multiple times on SFRA and the eventual refresh_token key might be same
-     * as that on PWA Kit which would incorrectly show both keys to be the same even though the sessions are different.
-     * @returns {boolean} true if the keys do not match (login state changed), false otherwise.
+     * When SFRA sends an auth token, it does so by splitting the JWT across
+     * multiple cookies to get around a character limit on ECOM cookie values.
+     * This method fetches the chunks and returns the combined token.
+     * @returns {string} access token or undefined
      */
-    private hasSFRAAuthStateChanged() {
-        const refreshTokenKey =
-            (this.get('refresh_token_registered') && 'refresh_token_registered') ||
-            'refresh_token_guest'
-
-        const refreshTokenCopyKey =
-            (this.get('refresh_token_registered_copy') && 'refresh_token_registered_copy') ||
-            'refresh_token_guest_copy'
-
-        if (DATA_MAP[refreshTokenKey].key !== DATA_MAP[refreshTokenCopyKey].key) {
-            return true
-        }
-
-        return this.get(refreshTokenKey) !== this.get(refreshTokenCopyKey)
+    private getSFRAAuthToken() {
+        const accessTokenChunk1 = this.get('access_token_sfra_1')
+        const accessTokenChunk2 = this.get('access_token_sfra_2')
+        const accessTokenChunk3 = this.get('access_token_sfra_3')
+
+        // Combines the access token chunks without any 'null' or 'undefined'
+        const accessToken = [accessTokenChunk1, accessTokenChunk2, accessTokenChunk3].join('')
```

Comment:
This seems brittle. Is there a way we could loop over these?

Reply:
This has been changed. Cookies as a whole now have support for being split over several chunks and looping is done to concatenate the chunks into a whole.